### PR TITLE
WIP openstack: Install openstackclient in tests image dynamically

### PR DESCRIPTION
**Pull request by dprince**
Update the openstack template so that it installs python-openstackclient
in the tests image dynamically.

### CODE REVIEW

File: ci-operator/templates/openshift/installer/cluster-launch-installer-openstack-e2e.yaml [hunk 1]
```diff
@@ -184,7 +220,9 @@ objects:
           exit 0
         }
 
-        ${TEST_COMMAND}
+        #FIXME: hard wire this to run parallel tests
```
Comment (dprince) [inline ci-operator/templates/openshift/installer/cluster-launch-installer-openstack-e2e.yaml]: NOTE: this will get removed once I see if this works. I wanted to hard wire it here as we don't run full tests in most places yet (just the canary jobs)

File: ci-operator/templates/openshift/installer/cluster-launch-installer-openstack-e2e.yaml [hunk 2]
```diff
@@ -184,7 +220,9 @@ objects:
...
+        #${TEST_COMMAND}
+        TEST_SUITE=openshift/conformance/parallel run-tests
```
Comment (wking) [inline ci-operator/templates/openshift/installer/cluster-launch-installer-openstack-e2e.yaml]: Doesn't seem like part of this PR.  Can it move into a separate PR if you feel like there's a reason we can't delegate the choice of tests to the job definition?

File: ci-operator/templates/openshift/installer/cluster-launch-installer-openstack-e2e.yaml [hunk 3]
```diff
@@ -124,6 +124,42 @@ objects:
...
+        cat > /etc/yum.repos.d/rdo-stein.repo <<-"EOF_CAT"
+        [openstack-stein]
+        name=OpenStack Stein Repository
+        baseurl=http://mirror.centos.org/centos/7/cloud/$basearch/openstack-stein/
+        gpgcheck=1
+        enabled=1
+        gpgkey=file:///etc/pki/rpm-gpg/RPM-GPG-KEY-CentOS-SIG-Cloud
+        EOF_CAT
+
+        cat > /etc/pki/rpm-gpg/RPM-GPG-KEY-CentOS-SIG-Cloud <<-"EOF_CAT"
```
Comment (wking) [inline ci-operator/templates/openshift/installer/cluster-launch-installer-openstack-e2e.yaml]: nit: `<<EOF` (or `EOF_CAT`, if you want) is sufficient, because you don't use the tab-indents that `<<-` enables.

File: ci-operator/templates/openshift/installer/cluster-launch-installer-openstack-e2e.yaml (hunk 2 above)
Comment (dprince) [reply ci-operator/templates/openshift/installer/cluster-launch-installer-openstack-e2e.yaml]: See my note above. My intent was to remove this once I see it pass. The issue is the OpenStack test jobs have TEST_COMMAND set to something that runs only a minimal set of smoke tests. This needs to change over time I think but as is now I just had this FIXME here to test it real quick in CI.

File: ci-operator/templates/openshift/installer/cluster-launch-installer-openstack-e2e.yaml (hunk 3 above)
Comment (dprince) [reply ci-operator/templates/openshift/installer/cluster-launch-installer-openstack-e2e.yaml]: Ack, I can remove the quoted EOF_CAT's

File: ci-operator/templates/openshift/installer/cluster-launch-installer-openstack-e2e.yaml [hunk 4]
```diff
@@ -124,6 +124,42 @@ objects:
...
+        -----BEGIN PGP PUBLIC KEY BLOCK-----
+        Version: GnuPG v2.0.22 (GNU/Linux)
+         
+        mQENBFVWcCcBCACfm3eQ0526/I0/p7HpR0NjK7K307XHhnbcbZv1sDUjQABDaqh0
```
Comment (wking) [inline ci-operator/templates/openshift/installer/cluster-launch-installer-openstack-e2e.yaml]: For future reference, this is:

```console
$ gpg --with-fingerprint cert-extracted-from-template.pub
pub  2048R/764429E6 2015-05-15 CentOS Cloud SIG (http://wiki.centos.org/SpecialInterestGroup/Cloud) <security@centos.org>
      Key fingerprint = 736A F511 6D9C 40E2 AF6B  074B F9B9 FEE7 7644 29E6
```

from [here][1].

[1]: https://www.centos.org/keys/#cloud-sig

File: ci-operator/templates/openshift/installer/cluster-launch-installer-openstack-e2e.yaml [hunk 5]
```diff
@@ -124,6 +124,42 @@ objects:
 
         mkdir -p "${HOME}"
 
+        # Install python-openstackclient which is currently required to run
+        # some tests (in parallel mode). See: https://github.com/kubernetes/kubernetes/blob/master/test/e2e/storage/drivers/in_tree.go#L956
+        cat > /etc/yum.repos.d/rdo-stein.repo <<-"EOF_CAT"
```
Comment (wking) [inline ci-operator/templates/openshift/installer/cluster-launch-installer-openstack-e2e.yaml]: From [CI][1]:

    /bin/bash: line 12: /etc/yum.repos.d/rdo-stein.repo: Permission denied

I'm not clear on why.

[1]: https://prow.svc.ci.openshift.org/view/gcs/origin-ci-test/pr-logs/pull/openshift_release/4933/rehearse-4933-pull-ci-openshift-cluster-api-provider-openstack-master-e2e-openstack/1#0:build-log.txt%3A27

File: ci-operator/templates/openshift/installer/cluster-launch-installer-openstack-e2e.yaml [hunk 6]
```diff
@@ -124,6 +124,44 @@ objects:
 
         mkdir -p "${HOME}"
 
+        # Install python-openstackclient which is currently required to run
```
Comment (smarterclayton) [inline ci-operator/templates/openshift/installer/cluster-launch-installer-openstack-e2e.yaml]: Create a function for this for better encapsulation

File: ci-operator/templates/openshift/installer/cluster-launch-installer-openstack-e2e.yaml [hunk 7]
```diff
@@ -124,6 +124,44 @@ objects:
...
+        # some tests (in parallel mode). See: https://github.com/kubernetes/kubernetes/blob/master/test/e2e/storage/drivers/in_tree.go#L956
+        sudo bash <<EOF_SUDO
```
Comment (smarterclayton) [inline ci-operator/templates/openshift/installer/cluster-launch-installer-openstack-e2e.yaml]: You probably won’t have sudo

File: ci-operator/templates/openshift/installer/cluster-launch-installer-openstack-e2e.yaml [hunk 8]
```diff
@@ -124,6 +124,44 @@ objects:
...
+        uWrW/wU0efz0aDB5c4NPg/yXfNvujvlda8ADJwZXVBQphzvaIKwl4PqBsEnxC10I
+        93T/0iyphAhfMRJ5R8AbEHMj7uF+TWTX/JoyQagllMqWTwoP4DFRutPdOmmjwvSV
+        kWItH7hq6z9+M4dhlqeoOvPbL5oCxX7TVmLck02Q5gI4syULOa7sqntzUQKFkhWp
+        9U0+5KrBQBKezrurrrkq/WZR3WNE1KQfNQ77f7S2JcXJdOaKgJ7xe7Y2flPq98Aq
+        wKXK7l1c3dc=
+        =W6yF
+        -----END PGP PUBLIC KEY BLOCK-----
+        EOF_CAT
+
+        yum install -y --setopt=tsflags=nodocs python-openstackclient
```
Comment (smarterclayton) [inline ci-operator/templates/openshift/installer/cluster-launch-installer-openstack-e2e.yaml]: I doubt you’re going to have sufficient perms for this

File: ci-operator/templates/openshift/installer/cluster-launch-installer-openstack-e2e.yaml (hunk 7 above)
Comment (wking) [reply ci-operator/templates/openshift/installer/cluster-launch-installer-openstack-e2e.yaml]: The permissions issues are making `pip install --user ...` look more appealing ;)

File: ci-operator/templates/openshift/installer/cluster-launch-installer-openstack-e2e.yaml [hunk 9]
```diff
@@ -124,6 +124,44 @@ objects:
 
         mkdir -p "${HOME}"
 
+        # Install python-openstackclient which is currently required to run
+        # some tests (in parallel mode). See: https://github.com/kubernetes/kubernetes/blob/master/test/e2e/storage/drivers/in_tree.go#L956
+        sudo bash <<EOF_SUDO
+        cat > /etc/yum.repos.d/rdo-stein.repo <<-EOF_CAT
+        [openstack-stein]
+        name=OpenStack Stein Repository
+        baseurl=http://mirror.centos.org/centos/7/cloud/$basearch/openstack-stein/
```
Comment (mandre) [inline ci-operator/templates/openshift/installer/cluster-launch-installer-openstack-e2e.yaml]: `$basearch` needs to be escaped.

File: ci-operator/templates/openshift/installer/cluster-launch-installer-openstack-e2e.yaml (hunk 7 above)
Comment (dprince) [reply ci-operator/templates/openshift/installer/cluster-launch-installer-openstack-e2e.yaml]: yeah, but like I mentioned on slack we don't have 'pip' either. I have to Yum install something.

**Comment by dprince** [comment]
/hold

Clearly installing RPMs in the container dynamically will fail without bumping the security context or adding sudo privs. So I think this is a bust.

I'd still like to explore the future possibility of using an OpenStack specific contain layered on top of the 'test' container which could contain these RPMs which seems the cleanest way to accommodate what I'm trying to do here.

I'm going to work up a separate PR which does the pip approach... stay tuned.
